### PR TITLE
feat: [Tree/TreeSelect] support A11y aria

### PR DESCRIPTION
**Pull request by boomboomchen**
[中文模板 / Chinese Template](https://github.com/DouyinFE/semi-design/blob/main/.github/PULL_REQUEST_TEMPLATE.zh-CN.md)

- [ ] I have read and followed [Pull Request Guidelines](https://github.com/DouyinFE/semi-design/blob/main/CONTRIBUTING-en-US.md#pull-request-guidelines) of the contributing guide.

### What kind of change does this PR introduce? (check at least one)

 - [ ] Bugfix
 - [ ] Feature
 - [ ] Code style update
 - [ ] Refactor
 - [ ] Test Case
 - [ ] TypeScript definition update
 - [ ] Document improve
 - [ ] CI/CD improve
 - [ ] Branch sync
 - [ ] Other, please describe:

### PR description

Fixes #

### Changelog
🇨🇳 Chinese
- feat: Tree 支持 A11y aria
- feat: TreeSelect 支持 A11y aria

---

🇺🇸 English
- feat: Tree supports A11y aria
- feat: TreeSelect supports A11y aria

### Checklist
- [ ] Test or no need
- [ ] Document or no need
- [ ] Changelog or no need

### CODE REVIEW

**Comment by pointhalo** [comment]
@boomboomchen 我这边也改了treeSelect与tree，相关修改见：https://github.com/DouyinFE/semi-design/pull/479  
可以先解决一下冲突。

File: packages/semi-ui/tree/treeNode.tsx [hunk 1]
```diff
@@ -242,9 +266,9 @@ export default class TreeNode extends PureComponent<TreeNodeProps, TreeNodeState
         });
         return (
             <ul className={wrapperCls}>
-                <span className={`${prefixcls}-label ${prefixcls}-label-empty`}>
+                <li className={`${prefixcls}-label ${prefixcls}-label-empty`}>
```
Comment (shijiatongxue) [inline packages/semi-ui/tree/treeNode.tsx:269]: 这里记得写在 changelog

File: packages/semi-ui/tree/treeNode.tsx [hunk 2]
```diff
@@ -201,8 +220,13 @@ export default class TreeNode extends PureComponent<TreeNodeProps, TreeNodeState
         const { checked, halfChecked } = this.props;
         const disabled = this.isDisabled();
         return (
-            <div onClick={this.onCheck}>
+            <div
```
Comment (shijiatongxue) [inline packages/semi-ui/tree/treeNode.tsx:223]: 199 行，折叠按钮需要加个 role 和 aria-label

File: packages/semi-ui/treeSelect/index.tsx [hunk 3]
```diff
@@ -914,25 +933,30 @@ class TreeSelect extends BaseComponent<TreeSelectProps, TreeSelectState> {
...
          * Reasons for disabling the a11y eslint rule:
          * The following attributes(aria-controls,aria-expanded) will be automatically added by Tooltip, no need to declare here
          */
         return (
             <div
+                // eslint-disable-next-line jsx-a11y/role-has-required-aria-props
+                role='combobox'
+                aria-label='TreeSelect input box'
+                aria-disabled={disabled}
+                aria-haspopup="listbox"
```
Comment (shijiatongxue) [inline packages/semi-ui/treeSelect/index.tsx:947]: 这里改成 tree？

File: packages/semi-ui/tree/treeNode.tsx [hunk 4]
```diff
@@ -376,9 +400,14 @@ export default class TreeNode extends PureComponent<TreeNodeProps, TreeNodeState
         return (
             <li
                 className={nodeCls}
-                role="treenode"
+                role="treeitem"
```
Comment (shijiatongxue) [inline packages/semi-ui/tree/treeNode.tsx:403]: aria-setsize
aria-posinset 也加下吧

参考：https://www.w3.org/TR/wai-aria-practices/examples/treeview/treeview-1/treeview-1b.html

File: packages/semi-ui/tree/index.tsx [hunk 5]
```diff
@@ -739,9 +741,9 @@ class Tree extends BaseComponent<TreeProps, TreeState> {
                     labelEllipsis: typeof labelEllipsis === 'undefined' ? virtualize : labelEllipsis,
                 }}
             >
-                <div className={wrapperCls} role="listbox" style={style}>
+                <div aria-label={this.props['aria-label']} className={wrapperCls} style={style}>
                     {filterTreeNode ? this.renderInput() : null}
-                    <div className={listCls} role="tree">
+                    <div className={listCls} role={noData? 'none' : 'tree'}>
```
Comment (shijiatongxue) [inline packages/semi-ui/tree/index.tsx:746]: 如果多选应该加上 `aria-multiselectable`

https://www.w3.org/TR/wai-aria-1.1/#aria-multiselectable

File: packages/semi-ui/tree/treeNode.tsx (hunk 1 above)
Comment (boomboomchen) [reply packages/semi-ui/tree/treeNode.tsx:269]: 好的，我补下。这里是因为ul children 为 span不符合规范，a11y插件会有警告，看了下不影响样式

File: content/input/treeselect/index.md [hunk 6]
```diff
@@ -1235,6 +1235,16 @@ function Demo() {
...
 | isLeaf| 是否为叶子节点 | boolean |-|
 
+
+## Accessibility
+
+### Aria
+
+- TreeSelect 支持传入 `aria-label`、`aria-describedby`、`aria-errormessage`、`aria-invalid`、`aria-labelledby`、`aria-required` 来表示该 TreeSelect 作用;
+- TreeSelect 会为每个子节点分别设置 `aria-disabled`、`aria-checked`、`aria-selected`、`aria-level` 来表明节点状态及层级;
+- TreeSelect 支持通过按下 Enter 键来展开下拉框以及多选时选中节点。
```
[the remark this answers is not in the document]
Comment (boomboomchen) [reply content/input/treeselect/index.md:1245]: 嗯嗯，我先删掉，等后续键盘事件写齐全了再加

File: packages/semi-ui/treeSelect/index.tsx (hunk 3 above)
Comment (boomboomchen) [reply packages/semi-ui/treeSelect/index.tsx:947]: 感觉这里改成tree好像不太合适？这里render的selection，它是TreeSelect的trigger。而Tree是TreeSelect弹窗的一部分。

Comment (shijiatongxue) [reply packages/semi-ui/treeSelect/index.tsx:947]: haspopup 那里改成 tree 合适不

Comment (boomboomchen) [reply packages/semi-ui/treeSelect/index.tsx:947]: 嗯嗯可以的，我改一下

**Comment by boomboomchen** [comment]
@shijiatongxue 已修改，帮忙再看下～

**Pull request merged**